### PR TITLE
Remove Rule board from docs

### PR DESCRIPTION
**Pull request by s-santillan**
# Thanks for improving Semgrep Docs 😀

### Please ensure

(N/A) A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs
- [x] If you have changed any header tag links (doc/#this-kind-of-anchor), update all instances of that link

----

 - Update all Rule board links because the endpoint is about to change (`.../board/` to `.../policies/` - see PR below)
- Update associated text (procedures, description of policies) because the flow is slightly different for various tasks.
- Set up redirect for docs-related links (`semgrep-code/rule-board` to `semgrep-code/policies/`) in docusaurus.config.js
- Retained mentions of rule board in Release notes.

For reviewers: When I rewrote the procedures, I used the labels on UI elements for the upcoming [Semgrep App PR 9426](https://github.com/returntocorp/semgrep-app/pull/9426) merge.

I also went through a bunch of files, so I cleaned up some stuff:

~~- "Ignoring files, folders, and code" was renamed to "Ignoring files and folders", because the `nosemgrep` annotation actually generates a finding and then ignores it as a triage action, as opposed to "excluding file from scan" which is what Semgrep does for files and folders. The distinction was added to a callout box in Ignoring files and folders but we never updated the title.~~ **I am walking back this decision because I am conflicted about this change because the doc itself provides information on ignoring parts of code through nosemgrep, and it feels natural for the user to search for ignoring code. What I can do is stress the difference between ignoring as skip and ignoring as triage.

Preferred term to refer to Policies should be **Policies page** in most cases or **Policies** if the meaning is clear. ([Definition of page, dialog and other UI elements in Google Style Guide](https://developers.google.com/style/ui-elements#term-windows))

### CODE REVIEW

**Comment by s-santillan** [comment]
PR on code side: [https://github.com/returntocorp/semgrep-app/pull/9426](https://github.com/returntocorp/semgrep-app/pull/9426)

**Review by pabloest — COMMENTED** [review]
Does this page need to be deleted? I saw the redirect for it.

CleanShot 2023-07-28 at 14 29 56@2x

**Comment by s-santillan** [comment]
@pabloest Yep good catch I'll remove it!

**Comment by s-santillan** [comment]
> Does this page need to be deleted? I saw the redirect for it.
> 

Removed!

**Comment by pabloest** [comment]
@s-santillan I think the contents of the PR are good now, however there is one test failing and I’m not sure why.

File: docs/semgrep-ci/running-semgrep-ci-without-semgrep-cloud-platform.md [hunk 1]
```diff
@@ -52,8 +52,8 @@ By configuring a job's parameters, you are able to achieve the following goals:
...
 
-* Stand-alone CI jobs cannot send [PR or MR comments](/semgrep-cloud-platform/github-pr-comments). These comments describe the finding and help developers resolve vulnerabilities and other code issues.
-* Stand-alone CI jobs cannot block a pull or merge request based on a finding. There is no distinction or tiering between rules. Alternatively, Semgrep Cloud Platform's [Rule Board](/semgrep-code/rule-board) can tier rules and undertake actions, such as blocking PRs or MRs on a finding generated by a reliable rule.
+* Stand-alone CI jobs cannot send [PR comments]((/semgrep-cloud-platform/github-pr-comments) or [MR comments](/semgrep-cloud-platform/gitlab-mr-comments). These comments describe the finding and help developers resolve vulnerabilities and other code issues.
+* Stand-alone CI jobs cannot block a pull or merge request based on a finding. There are no user-defined [rule modes] to distinguish between rules. Alternatively, Semgrep Cloud Platform's [Policies page](/semgrep-code/policies) lets users define certain actions, such as blocking PRs or MRs on a finding generated by a reliable rule.
```
Comment (armchairlinguist) [inline docs/semgrep-ci/running-semgrep-ci-without-semgrep-cloud-platform.md:56]: Is there supposed to be a link with the rule modes text?

File: src/components/procedure/_remove-ruleset.mdx [hunk 2]
```diff
@@ -1,20 +1,10 @@
...
+3. Click the <i class="fas fa-check-square"></i> **Matching rules**.
+4. Click **Edit > Disabled**.
 
 :::info
-- Individual rules within rulesets can only be disabled, not removed from the Rule board. To disable an individual rule in **[Rule board](https://semgrep.dev/orgs/-/board)**, click the <i class="fa-solid fa-toggle-large-on"></i> toggle to disable the rule. See also [Disabling rules](/semgrep-code/rule-board/#disabling-rules).
-- When you remove a rule from the Rule Board, all associated findings on Findings page and Dashboard page are removed also.
+- When you disable a rule, all previous findings generated by that rule are still retained.
 :::
-
-You can also remove a rule on the Findings page, to do so, follow these steps:
```
Comment (armchairlinguist) [inline src/components/procedure/_remove-ruleset.mdx:13]: This is still possible if you don't use group-by-rule - is there a reason it was removed?

**Review by armchairlinguist — COMMENTED** [review]
I just had one significant comment - I think some functionality that still exists may have had its instructions removed, unless they just got put somewhere else and I didn't notice.

Also, this change updates names of UI elements, but in the UI the rule board is still "Board". Will this release be coordinated with changing the UI name?

File: docs/semgrep-ci/running-semgrep-ci-without-semgrep-cloud-platform.md (hunk 1 above)
Comment (s-santillan) [reply docs/semgrep-ci/running-semgrep-ci-without-semgrep-cloud-platform.md:56]: Good catch, I'll add it in.

File: src/components/procedure/_remove-ruleset.mdx (hunk 2 above)
Comment (s-santillan) [reply src/components/procedure/_remove-ruleset.mdx:13]: Thanks for catching that. I'll update the docs here :)

**Comment by s-santillan** [comment]
Hi @armchairlinguist ! Here's how I addressed your notes:

> I just had one significant comment - I think some functionality that still exists may have had its instructions removed, unless they just got put somewhere else and I didn't notice.

Yup, added this back in after testing. Thanks for the catch.

> Also, this change updates names of UI elements, but in the UI the rule board is still "Board". Will this release be coordinated with changing the UI name?

Currently In the navbar/sidenav, it's Board; in the page's Header itself, it's called Policies. The code-side PR I was supposed to sync this to is not yet merged with the navbar correction and I'd rather unblock these changes so we can merge now and update the docs with a follow up PR, including correct screenshots.

I updated various endpoints back from /policies to /board for this reason as well.

**Review by armchairlinguist — APPROVED** [review]

**Pull request merged**